### ISSUE TITLE
R65 Support

### ISSUE THREAD

**Issue by Sachin9328** [rnayabed/rangoli#76]
Program doesn't support Royal Kludge R65, it shows connect a Royal Kludge Keyboard to get started.

**Comment by sfzaw** [comment]
Not sure if this going to help the dev or not.
http://en.rkgaming.com/download/12/


RK-R65 QMK Software
Version: wired
Attribute: RGB
VIA web-based software and a tutorial for QMK flashing
Download: [windows](http://en.rkgaming.com/media/files/20240126/20240126160433_2716.zip)

**Comment by MrunmayS** [comment]
> Not sure if this going to help the dev or not. http://en.rkgaming.com/download/12/
> 
> RK-R65 QMK Software Version: wired Attribute: RGB VIA web-based software and a tutorial for QMK flashing Download: [windows](http://en.rkgaming.com/media/files/20240126/20240126160433_2716.zip)

Have you tried the software ?

**Comment by sfzaw** [comment]
> Have you tried the software ?

No. its windows software. The link I gave is from the official royal kludge website.  The keyboard is not available yet in my country. However, I am planning to buy r65 or r75 soon enough.  Official Amazon provides us with this link for software 

bit.ly/RKRGBSOFT
https://drive.google.com/drive/folders/1_oio6FGBsqBJ6brhvmD_hbdn9HDlTlOL

**Comment by MrunmayS** [comment]
> > Have you tried the software ?
> 
> No. its windows software. The link I gave is from the official royal kludge website. The keyboard is not available yet in my country. However, I am planning to buy r65 or r75 soon enough. Official Amazon provides us with this link for software
> 
> bit.ly/RKRGBSOFT https://drive.google.com/drive/folders/1_oio6FGBsqBJ6brhvmD_hbdn9HDlTlOL

I shall try and let you know how it works. thanks for sharing the link to software. The place i got the keyboard from didnt have the link.

**Comment by sfzaw** [comment]
> I shall try and let you know how it works. thanks for sharing the link to software. The place i got the keyboard from didnt have the link.

Theres a R65 version with QMK that will use Westberry MCU. But I suspect there are also R65 versions without QMK so be careful. Before trying, match the PID and vendor ID of the USB device with the json file(provided in the zip). there's a very high chance that r65 will be a tri-mode version
 
 P.S. Check vendor ID if it's from Sinowealth (258a). if so, it may get supported by Rangoli with some minor tweaking.

**Comment by xjunko** [comment]
I use R65 myself, VID is indeed `258a`
```bash
Bus 001 Device 003: ID 258a:0172 BY Tech USB Keyboard
```

**Comment by Gtocss** [comment]
for mac wts the solution?
